tests/keras/layers/recurrent_test.py

```diff
@@ -165,6 +165,90 @@ def test_masking_correctness(layer_class):
     assert_allclose(out7, out6, atol=1e-5)
 
 
+@pytest.mark.skipif(K.backend() == 'cntk', reason='Not supported.')
+def test_masking_correctness_output_not_equal_to_first_state():
+
+    class Cell(keras.layers.Layer):
+
+        def __init__(self):
+            self.state_size = None
+            self.output_size = None
+            super(Cell, self).__init__()
+
+        def build(self, input_shape):
+            self.state_size = input_shape[-1]
+            self.output_size = input_shape[-1]
+
+        def call(self, inputs, states):
+            return inputs, [s + 1 for s in states]
+
+    for unroll in [True, False]:
```

Comment:
I think you can also here test with random values, but check that the numpy arrays given by `model.predict` are the same   for both cases.